src/OrchardCore/OrchardCore.Abstractions/Shell/IShellContextReleaseService.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Environment.Shell;
+
+public interface IShellContextReleaseService
+{
+    /// <summary>
+    /// Adds a pending request to release the shell upon completion of the current HTTP-request.
+    /// </summary>
+    void RequestRelease();
+
+    /// <summary>
+    /// It suspends the pending release request to ensure that the shell remains intact when 'ProcessAsync()' is invoked.
+    /// </summary>
+    void SuspendReleaseRequest();
```

Comment:
This cancels the release (i.e. it won't do it) rather than suspends it (won't do it now but will do it at a later time).

Reply:
I think suspend is the the correct term to use. Suspending something means it could be resumed later where cancel means stop it completely. In this case we suspend, but then we could later request it again.

Reply:
Wouldn't it need a Resume operation, then?

Reply:
Not really because you don't know if is was already suspended. I think Request and Suspend are cleaner either you request or suspend.